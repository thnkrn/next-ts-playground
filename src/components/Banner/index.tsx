import { styled } from '@compiled/react'
import Icon from 'components/Common/Icon'
import { ICONS } from 'constants/icons'
import TYPO from 'constants/typography'
import Image from 'next/image'

interface Props {
  isPhone: boolean
}

const StyledProfileWrapper = styled.div`
  @media screen and (max-width: 480px) {
    flex-direction: column;
    margin-top: 0;
  }

  display: flex;
  align-items: center;
  margin-top: 2vw;
`

const StyledLocation = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  margin-top: 15px;
`

const StyledProfileImage = styled.div`
  @media screen and (max-width: 480px) {
    width: 50vw;
  }

  width: 25vw;
  display: flex;
  flex-direction: column;
`

const StyledImage = styled(Image)`
  border-radius: 50%;
`

const StyledTextWrapper = styled.div`
  @media screen and (max-width: 480px) {
    margin: 15px 0 0 0;
  }

  display: flex;
  flex-direction: column;
  margin-left: 5vw;

  h1 {
    margin-top: 10px;
  }

  span {
    color: #88b1b3;
  }
`

const StyledPosition = styled.div`
  @media screen and (max-width: 480px) {
    display: flex;
    flex-direction: column;
    justify-content: center;
    margin-top: 10px;
  }

  margin-top: 5px;
`

const StyledPositionTitle = styled.span`
  @media screen and (max-width: 480px) {
    margin-left: 0;
  }

  margin-left: 10px;
`

const Banner = ({ isPhone }: Props) => (
  <StyledProfileWrapper>
    <StyledProfileImage>
      <StyledImage
        src="/images/profile.jpeg"
        alt="Picture of thnkrn"
        width={1385}
        height={1385}
        sizes={isPhone ? '50vw' : '35vw'}
        layout="responsive"
        quality="100"
      />
      <StyledLocation>
        <Icon src={ICONS.location} />
        <span className={TYPO.body2} style={{ marginLeft: '5px' }}>
          Bangkok, Thailand
        </span>
      </StyledLocation>
    </StyledProfileImage>
    <StyledTextWrapper>
      <span className={TYPO.body1}>Hi, I'm</span>
      <h1 className={TYPO.h1}>Thanakorn Ariyagusolsuthi</h1>
      <StyledPosition>
        <StyledPositionTitle className={TYPO.italic2}>
          Assistant Manager, Software Engineer
        </StyledPositionTitle>
      </StyledPosition>
    </StyledTextWrapper>
  </StyledProfileWrapper>
)

export default Banner
