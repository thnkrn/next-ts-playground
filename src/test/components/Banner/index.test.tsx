import Banner from 'components/Banner'
import { render } from 'test/util'

describe('Banner', () => {
  describe('Rendering', () => {
    it('should match snapshot if it is phone', () => {
      const { asFragment } = render(<Banner isPhone />)

      expect(asFragment()).toMatchSnapshot()
    })

    it('should match snapshot if it is desktop', () => {
      const { asFragment } = render(<Banner isPhone={false} />)

      expect(asFragment()).toMatchSnapshot()
    })
  })
})
