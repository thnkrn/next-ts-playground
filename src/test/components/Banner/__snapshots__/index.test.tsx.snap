// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner Rendering should match snapshot if it is desktop 1`] = `
<DocumentFragment>
  <style
    data-cmpld="true"
  >
    ._1e0c1txw{display:flex}._4cvr1h6o{align-items:center}._1bah1h6o{justify-content:center}._19pk9lop{margin-top:2vw}@media screen and (max-width:480px){._46cc1bp4{flex-direction:column}._za9yidpf{margin-top:0}}
  </style>
  <div
    class="_46cc1bp4 _za9yidpf _1e0c1txw _4cvr1h6o _1bah1h6o _19pk9lop"
  >
    <style
      data-cmpld="true"
    >
      ._1bsb1jyv{width:25vw}._2lx21bp4{flex-direction:column}._hu4x1wqb .image{border-radius:60px}@media screen and (max-width:480px){._18l8a4lr{width:50vw}}
    </style>
    <div
      class="_18l8a4lr _1bsb1jyv _1e0c1txw _2lx21bp4 _hu4x1wqb"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 136.03515625% 0px 0px 0px;"
        />
        <img
          alt="Picture of thnkrn"
          class="image"
          data-nimg="responsive"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; filter: blur(20px); background-size: cover; background-image: url(data:image/svg+xml;base64,Cjxzdmcgd2lkdGg9IjEwMjQiIGhlaWdodD0iMTM5MyIgdmVyc2lvbj0iMS4xIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHhtbG5zOnhsaW5rPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5L3hsaW5rIj4KICA8ZGVmcz4KICAgIDxsaW5lYXJHcmFkaWVudCBpZD0iZyI+CiAgICAgIDxzdG9wIHN0b3AtY29sb3I9IiMzMzMiIG9mZnNldD0iMjAlIiAvPgogICAgICA8c3RvcCBzdG9wLWNvbG9yPSIjMjIyIiBvZmZzZXQ9IjUwJSIgLz4KICAgICAgPHN0b3Agc3RvcC1jb2xvcj0iIzMzMyIgb2Zmc2V0PSI3MCUiIC8+CiAgICA8L2xpbmVhckdyYWRpZW50PgogIDwvZGVmcz4KICA8cmVjdCB3aWR0aD0iMTAyNCIgaGVpZ2h0PSIxMzkzIiBmaWxsPSIjMzMzIiAvPgogIDxyZWN0IGlkPSJyIiB3aWR0aD0iMTAyNCIgaGVpZ2h0PSIxMzkzIiBmaWxsPSJ1cmwoI2cpIiAvPgogIDxhbmltYXRlIHhsaW5rOmhyZWY9IiNyIiBhdHRyaWJ1dGVOYW1lPSJ4IiBmcm9tPSItMTAyNCIgdG89IjEwMjQiIGR1cj0iMXMiIHJlcGVhdENvdW50PSJpbmRlZmluaXRlIiAgLz4KPC9zdmc+); background-position: 0% 0%;"
        />
        <noscript />
      </span>
      <style
        data-cmpld="true"
      >
        ._19pko7ao{margin-top:15px}
      </style>
      <div
        class="_1e0c1txw _4cvr1h6o _1bah1h6o _19pko7ao"
      >
        <style
          data-cmpld="true"
        >
          ._1bsb1tcg{width:24px}._4t3i1tcg{height:24px}._tzy4idpf{opacity:0}._18t5kb7n.loaded{opacity:1}
        </style>
        <svg
          class="_1bsb1tcg _4t3i1tcg _tzy4idpf _18t5kb7n icon loaded"
        >
          <use
            href="/icons/icons-sprite.svg#location"
          />
        </svg>
        <span
          class="body2"
          style="margin-left: 5px;"
        >
          Bangkok, Thailand
        </span>
      </div>
    </div>
    <style
      data-cmpld="true"
    >
      ._18u01l76{margin-left:5vw}._rzby19bv h1{margin-top:10px}._1tjq3eyv span{color:#88b1b3}@media screen and (max-width:480px){._za9yo7ao{margin-top:15px}._99s2idpf{margin-right:0}._183zidpf{margin-bottom:0}._wf1nidpf{margin-left:0}}
    </style>
    <div
      class="_za9yo7ao _99s2idpf _183zidpf _wf1nidpf _1e0c1txw _2lx21bp4 _18u01l76 _rzby19bv _1tjq3eyv"
    >
      <span
        class="body1"
      >
        Hi, I'm
      </span>
      <h1
        class="h1"
      >
        Thanakorn Ariyagusolsuthi
      </h1>
      <style
        data-cmpld="true"
      >
        ._19pk14y2{margin-top:5px}@media screen and (max-width:480px){._ygwp1txw{display:flex}._46cc1bp4{flex-direction:column}._1rkf1h6o{justify-content:center}._za9y19bv{margin-top:10px}}
      </style>
      <div
        class="_ygwp1txw _46cc1bp4 _1rkf1h6o _za9y19bv _19pk14y2"
      >
        <style
          data-cmpld="true"
        >
          ._18u019bv{margin-left:10px}@media screen and (max-width:480px){._wf1nidpf{margin-left:0}}
        </style>
        <span
          class="_wf1nidpf _18u019bv italic2"
        >
          Software Engineer
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Banner Rendering should match snapshot if it is phone 1`] = `
<DocumentFragment>
  <style
    data-cmpld="true"
  >
    ._1e0c1txw{display:flex}._4cvr1h6o{align-items:center}._1bah1h6o{justify-content:center}._19pk9lop{margin-top:2vw}@media screen and (max-width:480px){._46cc1bp4{flex-direction:column}._za9yidpf{margin-top:0}}
  </style>
  <div
    class="_46cc1bp4 _za9yidpf _1e0c1txw _4cvr1h6o _1bah1h6o _19pk9lop"
  >
    <style
      data-cmpld="true"
    >
      ._1bsb1jyv{width:25vw}._2lx21bp4{flex-direction:column}._hu4x1wqb .image{border-radius:60px}@media screen and (max-width:480px){._18l8a4lr{width:50vw}}
    </style>
    <div
      class="_18l8a4lr _1bsb1jyv _1e0c1txw _2lx21bp4 _hu4x1wqb"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 136.03515625% 0px 0px 0px;"
        />
        <img
          alt="Picture of thnkrn"
          class="image"
          data-nimg="responsive"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; filter: blur(20px); background-size: cover; background-image: url(data:image/svg+xml;base64,Cjxzdmcgd2lkdGg9IjEwMjQiIGhlaWdodD0iMTM5MyIgdmVyc2lvbj0iMS4xIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHhtbG5zOnhsaW5rPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5L3hsaW5rIj4KICA8ZGVmcz4KICAgIDxsaW5lYXJHcmFkaWVudCBpZD0iZyI+CiAgICAgIDxzdG9wIHN0b3AtY29sb3I9IiMzMzMiIG9mZnNldD0iMjAlIiAvPgogICAgICA8c3RvcCBzdG9wLWNvbG9yPSIjMjIyIiBvZmZzZXQ9IjUwJSIgLz4KICAgICAgPHN0b3Agc3RvcC1jb2xvcj0iIzMzMyIgb2Zmc2V0PSI3MCUiIC8+CiAgICA8L2xpbmVhckdyYWRpZW50PgogIDwvZGVmcz4KICA8cmVjdCB3aWR0aD0iMTAyNCIgaGVpZ2h0PSIxMzkzIiBmaWxsPSIjMzMzIiAvPgogIDxyZWN0IGlkPSJyIiB3aWR0aD0iMTAyNCIgaGVpZ2h0PSIxMzkzIiBmaWxsPSJ1cmwoI2cpIiAvPgogIDxhbmltYXRlIHhsaW5rOmhyZWY9IiNyIiBhdHRyaWJ1dGVOYW1lPSJ4IiBmcm9tPSItMTAyNCIgdG89IjEwMjQiIGR1cj0iMXMiIHJlcGVhdENvdW50PSJpbmRlZmluaXRlIiAgLz4KPC9zdmc+); background-position: 0% 0%;"
        />
        <noscript />
      </span>
      <style
        data-cmpld="true"
      >
        ._19pko7ao{margin-top:15px}
      </style>
      <div
        class="_1e0c1txw _4cvr1h6o _1bah1h6o _19pko7ao"
      >
        <style
          data-cmpld="true"
        >
          ._1bsb1tcg{width:24px}._4t3i1tcg{height:24px}._tzy4idpf{opacity:0}._18t5kb7n.loaded{opacity:1}
        </style>
        <svg
          class="_1bsb1tcg _4t3i1tcg _tzy4idpf _18t5kb7n icon loaded"
        >
          <use
            href="/icons/icons-sprite.svg#location"
          />
        </svg>
        <span
          class="body2"
          style="margin-left: 5px;"
        >
          Bangkok, Thailand
        </span>
      </div>
    </div>
    <style
      data-cmpld="true"
    >
      ._18u01l76{margin-left:5vw}._rzby19bv h1{margin-top:10px}._1tjq3eyv span{color:#88b1b3}@media screen and (max-width:480px){._za9yo7ao{margin-top:15px}._99s2idpf{margin-right:0}._183zidpf{margin-bottom:0}._wf1nidpf{margin-left:0}}
    </style>
    <div
      class="_za9yo7ao _99s2idpf _183zidpf _wf1nidpf _1e0c1txw _2lx21bp4 _18u01l76 _rzby19bv _1tjq3eyv"
    >
      <span
        class="body1"
      >
        Hi, I'm
      </span>
      <h1
        class="h1"
      >
        Thanakorn Ariyagusolsuthi
      </h1>
      <style
        data-cmpld="true"
      >
        ._19pk14y2{margin-top:5px}@media screen and (max-width:480px){._ygwp1txw{display:flex}._46cc1bp4{flex-direction:column}._1rkf1h6o{justify-content:center}._za9y19bv{margin-top:10px}}
      </style>
      <div
        class="_ygwp1txw _46cc1bp4 _1rkf1h6o _za9y19bv _19pk14y2"
      >
        <style
          data-cmpld="true"
        >
          ._18u019bv{margin-left:10px}@media screen and (max-width:480px){._wf1nidpf{margin-left:0}}
        </style>
        <span
          class="_wf1nidpf _18u019bv italic2"
        >
          Software Engineer
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
